monitors/CRMonitor.js

```diff
@@ -0,0 +1,148 @@
+    const priceFeed = priceFunction(contractTime);
+    this.logger.debug({
+      at: "CRMonitor",
+      message: "Checking wallet collateralization radios",
+      price: priceFeed
+    });
+    // For each monitored wallet check if the current collaterlization ratio is below the monitored threshold.
+    // If it is, then send an alert of formatted markdown text.
+    for (let wallet of this.walletsToMonitor) {
+      const [shouldPush, crRatio] = this._shouldPushWalletNotification(wallet, priceFeed);
+      if (shouldPush) {
+        // Sample message:
+        // Risk alert: [Tracked wallet name] has fallen below [threshold]%.
+        // Current [name of identifier] value: [current identifier value].
+        const mrkdwn =
+          wallet.name +
+          " (" +
+          createEtherscanLinkMarkdown(this.web3, wallet.address) +
+          ") collateralization ratio has dropped to " +
+          this.formatDecimalString(crRatio) +
+          "% which is below the " +
+          wallet.crAlert +
+          "% threshold. Current value of " +
+          this.syntheticCurrencySymbol +
+          " is " +
+          this.formatDecimalString(priceFeed);
+
+        this.logger.info({
+          at: "ContractMonitor",
+          message: "Collateralization ratio alert 🚨!",
+          mrkdwn: mrkdwn
+        });
+      }
+    }
+  };
+
+  _getPositionInformation = address => {
+    const positionInfo = this.empClient.getAllPositions().filter(position => position.sponsor == address);
+    if (positionInfo.length == 0) {
+      return null;
+      // there should only ever be one position information object per address
+    } else return positionInfo[0];
+  };
+
+  _shouldPushWalletNotification(wallet, priceFeed) {
+    const positionInformation = this._getPositionInformation(wallet.address);
+    if (positionInformation == null) {
+      // There is no position information for the given wallet.
+      return [false, 0];
+    }
+
+    const collateral = positionInformation.amountCollateral;
+    const tokensOutstanding = positionInformation.numTokens;
+
+    // If the values for collateral or price have yet to resolve, dont push a notification
+    if (collateral == null || tokensOutstanding == null) {
+      return [false, 0];
+    }
+
+    // If CR = null then there are no tokens outstanding and so dont push a notification
+    const positionCR = this._calculatePositionCRPercent(collateral, tokensOutstanding, priceFeed);
+    if (positionCR == null) {
+      return [false, 0];
+    }
+
+    let shouldPushWalletNotification = false;
+    if (this._ltThreshold(positionCR, this.web3.utils.toWei(wallet.crAlert.toString()))) {
+      if (!this.walletsAlerted[wallet.address].crAlert) {
+        shouldPushWalletNotification = true;
+      }
+      this.walletsAlerted[wallet.address].crAlert = true;
+    } else {
+      this.walletsAlerted[wallet.address].crAlert = false;
+    }
+    return [shouldPushWalletNotification, positionCR];
```

Comment:
Stylistic question: why did you choose to return these as an array rather than an object? Not sure what the typical js style is for these sorts of situations. The alternative I've seen:
```
return {shouldPushWalletNotification, positionCR};
```

Is there a reason to use one over the other? (this doesn't actually matter, just genuinely curious)

Reply:
Ye there is a reason. If you encode it as an object as in your example then you need to put keys along with the value objects. This becomes a bit nasty however it's not too bad. I've updated it to be an object and catch it on the other side without requiring array structures.

Reply:
Doesn't matter to me -- just curious the reasoning. I've seen plenty of libraries use both styles.